ChopChop/Storage/StorageManager.swift

```diff
@@ -228,6 +229,167 @@ extension StorageManager {
+            name: recipe.name,
+            creator: userId,
+            servings: recipe.servings,
+            cuisine: cuisine,
+            difficulty: recipe.difficulty,
+            ingredients: ingredients,
+            steps: steps
+        )
+        try firebase.updateRecipeDetails(recipe: recipeRecord)
+    }
+
+    func removeRecipe(recipe: OnlineRecipe) throws {
+//        guard let id = recipe.id else {
+//            fatalError("Online recipe has no id.")
+//        }
+
+        try firebase.removeRecipe(recipeId: recipe.id)
+//        for rating in recipe.ratings {
+//            firebase.removeRating(ratingId: rating.ratingId)
+//        }
+        for rating in recipe.ratings {
+            firebase.removeUserRecipeRating(userId: rating.userId, rating: UserRating(recipeOnlineId: recipe.id, score: rating.score))
+        }
+    }
+
+    func addFollowee(userId: String, followeeId: String) {
+        firebase.addFollowee(userId: userId, followeeId: followeeId)
+    }
+
+    func removeFollowee(userId: String, followeeId: String) {
+        firebase.removeFollowee(userId: userId, followeeId: followeeId)
+    }
+
+    func rateRecipe(recipeId: String, userId: String, rating: RatingScore) throws {
+        firebase.addUserRecipeRating(userId: userId, rating: UserRating(recipeOnlineId: recipeId, score: rating))
+        firebase.addRecipeRating(onlineRecipeId: recipeId, rating: RecipeRating(userId: userId, score: rating))
+//        let ratingRecord = RatingRecord(userId: recipeId, recipeId: userId, rating: rating)
+//        try firebase.addRating(rating: ratingRecord)
+    }
+
+//    func unrateRecipe(ratingId: String) {
+//        firebase.removeRating(ratingId: ratingId)
+//    }
+    func unrateRecipe(recipeId: String, rating: RecipeRating) {
+        firebase.removeRecipeRating(onlineRecipeId: recipeId, rating: rating)
+        firebase.removeUserRecipeRating(userId: rating.userId, rating: UserRating(recipeOnlineId: recipeId, score: rating.score))
+    }
+
+//    func rerateRecipe(ratingId: String, newRating: RatingScore) {
+//        firebase.updateRating(ratingId: ratingId, rating: newRating)
+//    }
+    func rerateRecipe(recipeId: String, newRating: RecipeRating) {
+        firebase.updateRecipeRating(userId: newRating.userId, recipeId: recipeId, newScore: newRating.score)
+        firebase.updateUserRating(userId: newRating.userId, recipeId: recipeId, newScore: newRating.score)
+    }
+
+    func fetchAllFriends(userId: String) -> AnyPublisher<[User], Error> {
+        firebase.fetchFriendsId(userId: userId)
+            .flatMap({ followees in
+                firebase.fetchUsers(userId: followees)
+            })
+            .eraseToAnyPublisher()
+    }
+
+    //testing
+//    func fetchAllUsers(user: [String]) -> AnyPublisher<[User], Error> {
+//        firebase.fetchUsers(userId: user)
+//    }
+    func fetchAllPublishedRecipes(userId: String) -> AnyPublisher<[OnlineRecipe], Error> {
```

Comment:
fetchAllFolloweeRecipes alr covers this, don't know whether this method still needed

Reply:
This refers to self published recipe but i guess I can merge the two

Reply:
Actually it is different, fetchAllFolloweeRecipe is a two step process where u first fetch the id of all your followees then fetch all their recipes.